packages/grafana-flamegraph/src/FlameGraph/dataTransform.ts

```diff
@@ -67,7 +67,9 @@ export function nestedSetToLevels(container: FlameGraphDataContainer): [LevelIte
       children: [],
     };
 
-    if (uniqueLabels[container.getLabel(i)]) {
+    // if the label value is something like 'toString' then checking its position in the array will return 'ƒ toString() { [native code] }'
+    // which results in a TypeError when we try to push to it. So we also have to check if the array already has at least one value
+    if (uniqueLabels[container.getLabel(i)] && uniqueLabels[container.getLabel(i)].length > 0) {
```

Comment:
The proper fix I think here would be to change the `uniqueLabels` to `Map` so that string like `toString` actually works properly as that can be a valid name of a function in the profile.

Reply:
Removed for now and added to [backlog](https://github.com/orgs/grafana/projects/221/views/3?pane=issue&itemId=41909402) as it's really separate from this PR